src/Native/Runtime/arm/AllocFast.S

```diff
@@ -137,36 +137,36 @@ LEAF_ENTRY RhpNewArray, _TEXT
 LOCAL_LABEL(ArrayAlignSize):
         bic         r2, r2, #3
 
-        mov         r4, r0 // save EEType
-
+        push        {r0-r2}
```

Comment:
This will mess up stack alignment, and the unwind info will be wrong as well.

It would be better to push the extra registers in the prolog instead `PROLOG_PUSH "{r4-r6,lr}"`.